### PR TITLE
ROX-7875: Improve logging exceptions in qa-test

### PR DESCRIPTION
**Pull request by janisz**
## Description

Currently we are using only print/ln to print debug information. In most places we don't print stacktrace of exception so it's harder to debug issues. This PR introduce logger to our implementation classes. The drawback of this approach is that we may have much more logs now.

## Testing Performed

CI

### CODE REVIEW

File: qa-tests-backend/src/main/groovy/services/ApiTokenService.groovy [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package services
 
+import groovy.util.logging.Slf4j
```
Comment (gavin-stackrox) [inline qa-tests-backend/src/main/groovy/services/ApiTokenService.groovy:3]: Could this be done one the BaseService.groovy to apply to all Services or does the decorator have to happen on a class by class basis?

File: qa-tests-backend/src/main/groovy/util/Helpers.groovy [hunk 2]
```diff
@@ -130,13 +132,11 @@ class Helpers {
             proc.waitFor()
 
             if (proc.exitValue() != 0) {
-                println "Failed to scan the image."
-                println "Stderr: $serr"
-                println "Exit: ${proc.exitValue()}"
+                log.warn "Failed to scan the image.\nStderr: $serr\nExit: ${proc.exitValue()}"
```
Comment (porridge) [inline qa-tests-backend/src/main/groovy/util/Helpers.groovy:135]: Perhaps put exit code before the (possibly long) stderr dump?

File: qa-tests-backend/src/main/groovy/util/SplunkUtil.groovy [hunk 3]
```diff
@@ -127,13 +129,13 @@ class SplunkUtil {
                     .post("https://127.0.0.1:${port}/services/search/jobs")
         }
 
-        println response?.asString() //printout the response for debugging purposes
+        log.debug response?.asString() //printout the response for debugging purposes
```
Comment (porridge) [inline qa-tests-backend/src/main/groovy/util/SplunkUtil.groovy:132]:
```suggestion
        log.debug response?.asString()
```

File: qa-tests-backend/src/test/groovy/BaseSpecification.groovy [hunk 4]
```diff
@@ -402,10 +401,8 @@ class BaseSpecification extends Specification {
         return Env.get("IS_RACE_BUILD", null) == "true" || Env.CI_JOBNAME == "race-condition-tests"
     }
 
-    static Void printlnDated(String msg) {
-        def date = new Date()
-        def sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US)
-        println "${sdf.format(date)} ${msg}"
+    static void printlnDated(String msg) {
```
Comment (porridge) [inline qa-tests-backend/src/test/groovy/BaseSpecification.groovy:404]: Perhaps rename or even inline this method?

File: qa-tests-backend/src/main/groovy/services/ApiTokenService.groovy (hunk 1 above)
Comment (gavin-stackrox) [reply qa-tests-backend/src/main/groovy/services/ApiTokenService.groovy:3]: Nevermind. I see your comment on the other PR.

**Review by gavin-stackrox — APPROVED** [review]

File: qa-tests-backend/src/test/groovy/BaseSpecification.groovy (hunk 4 above)
Comment (janisz) [reply qa-tests-backend/src/test/groovy/BaseSpecification.groovy]: Inlined

**Pull request merged**